spec/jobs/submission_invoice_creation_job_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe SubmissionInvoiceCreationJob do
+  describe '#perform' do
+    let(:submission) { FactoryBot.create(:submission) }
+    let(:submit_invoice_request_mock) { Workday::SubmitCustomerInvoiceRequest.new(submission) }
+    before do
+      allow(Workday::SubmitCustomerInvoiceRequest).to receive(:new)
+        .with(submission).and_return(submit_invoice_request_mock)
```

Comment:
I found this difficult to follow and I'm not sure what is actually happening - it looks a bit like a snake eating itself with the `allow` stubbing `Workday::SubmitCustomerInvoiceRequest` to return the `let` which is also the value being `and_return`'d by the stub! Could these things be re-written, perhaps to make `submit_invoice_request_mock` a true double instead of an instance of `Workday:: SubmitCustomerInvoiceRequest`?

Reply:
Does this make more sense?

Reply:
👍🏽